prometheus/src/main/java/org/opensearch/sql/prometheus/request/PrometheusDescribeMetricRequest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+package org.opensearch.sql.prometheus.request;
+
+import static org.opensearch.sql.prometheus.data.constants.PrometheusFieldConstants.METRIC;
+import static org.opensearch.sql.prometheus.data.constants.PrometheusFieldConstants.TIMESTAMP;
+import static org.opensearch.sql.prometheus.data.constants.PrometheusFieldConstants.VALUE;
+
+import java.io.IOException;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.ToString;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.sql.data.type.ExprCoreType;
+import org.opensearch.sql.data.type.ExprType;
+import org.opensearch.sql.prometheus.client.PrometheusClient;
+
+/**
+ * Describe Metric meta data request.
+ */
+@ToString(onlyExplicitlyIncluded = true)
+public class PrometheusDescribeMetricRequest {
+
+  private final PrometheusClient prometheusClient;
+
+  @ToString.Include
+  private final String metricName;
```

Comment:
Could you add JavaDoc? or Use Optional<String>. I think two constructor make more sense in this case
* PrometheusDescribeMetricRequest(@Nonull PrometheusClient prometheusClient, @Nonull String metricName)
* PrometheusDescribeMetricRequest(@Nonull PrometheusClient prometheusClient)

Reply:
Will add JavaDoc and add optional for now. Will refactor the class if required when Utilizing this.